From ed5c0660ef2e60260001c1778870a21f36c81d61 Mon Sep 17 00:00:00 2001
From: Francois Cartegnie <fcvlcdev@free.fr>
Date: Mon, 28 Sep 2020 17:52:16 +0200
Subject: [PATCH 31/44] httpd: update activity time before processing timeout
 (#25151)

as connection state handling block has moved in
67cdaeea58641d6d515728596f8b194f52538e2b
sockets can now timeout when still active.
---
 src/network/httpd.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/network/httpd.c b/src/network/httpd.c
index 69e6399483..f4a7638477 100644
--- a/src/network/httpd.c
+++ b/src/network/httpd.c
@@ -1735,6 +1735,11 @@ static void httpdLoop(httpd_host_t *host)
                 break;
         }
 
+        if (val == 0) {
+            cl->i_activity_date = now;
+            delay = 0;
+        }
+
         if (cl->i_ref < 0 || (cl->i_ref == 0 &&
                     (cl->i_state == HTTPD_CLIENT_DEAD ||
                       (cl->i_activity_timeout > 0 &&
@@ -1745,11 +1750,6 @@ static void httpdLoop(httpd_host_t *host)
             continue;
         }
 
-        if (val == 0) {
-            cl->i_activity_date = now;
-            delay = 0;
-        }
-
         struct pollfd *pufd = ufd + nfd;
         assert (pufd < ufd + (sizeof (ufd) / sizeof (ufd[0])));
 
-- 
2.24.3 (Apple Git-128)

